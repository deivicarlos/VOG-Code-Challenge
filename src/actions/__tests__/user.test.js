import configureMockStore from 'redux-mock-store';
import {
  fetchUserInfoBegin,
  fetchUserInfoError,
  fetchUserInfoSuccess,
} from '../user';

const mockStore = configureMockStore();
const store = mockStore({});

describe('User action creators', () => {
  it('creates FETCH_USER_INFO_BEGIN before making API call', () => {
    // Dispatch the createSuccess action with the values of a new to-do.
    store.dispatch(fetchUserInfoBegin());
    expect(store.getActions()).toMatchSnapshot();
  });

  it('creates FETCH_USER_INFO_SUCCESS when api call is made successfully', () => {
    store.dispatch(fetchUserInfoSuccess());
    expect(store.getActions()).toMatchSnapshot();
  });

  it('creates FETCH_USER_INFO_ERROR when API call fails', () => {
    // Dispatch the createSuccess action with the values of a new to-do.
    store.dispatch(fetchUserInfoError());
    expect(store.getActions()).toMatchSnapshot();
  });
});
