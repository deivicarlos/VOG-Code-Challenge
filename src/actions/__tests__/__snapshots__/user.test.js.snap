// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User action creators creates FETCH_USER_INFO_BEGIN before making API call 1`] = `
Array [
  Object {
    "type": "FETCH_USER_INFO_BEGIN",
  },
]
`;

exports[`User action creators creates FETCH_USER_INFO_ERROR when API call fails 1`] = `
Array [
  Object {
    "type": "FETCH_USER_INFO_BEGIN",
  },
  Object {
    "type": "FETCH_USER_INFO_SUCCESS",
    "user": undefined,
  },
  Object {
    "type": "FETCH_USER_INFO_ERROR",
  },
]
`;

exports[`User action creators creates FETCH_USER_INFO_SUCCESS when api call is made successfully 1`] = `
Array [
  Object {
    "type": "FETCH_USER_INFO_BEGIN",
  },
  Object {
    "type": "FETCH_USER_INFO_SUCCESS",
    "user": undefined,
  },
]
`;
